packages/core/src/browser/widget-open-handler.ts

```diff
@@ -86,16 +86,17 @@ export abstract class WidgetOpenHandler<W extends BaseWidget> implements OpenHan
      * @returns promise of the widget that resolves when the widget has been opened.
      */
     async open(uri: URI, options?: WidgetOpenerOptions): Promise<W> {
-        const widget = await this.getOrCreateWidget(uri, options);
-        await this.doOpen(widget, options);
+        const isWidgetExisting = this.isWidgetExisting(uri, options);
+        const widget = isWidgetExisting ? (await this.getWidget(uri, options))! : await this.createWidget(uri, options);
+        await this.doOpen(widget, isWidgetExisting, options);
```

Comment:
I'm not sure I understand the changes, shouldn't we call `getOrCreateWidget` which already checks if the widget exists, and if not creates it?

- https://github.com/eclipse-theia/theia/pull/8913/files#diff-d8c57e795e821606a1336d37fda1835453c6f572e9dc17e99f7ab10e65096fffR228-R235

We are duplicating functionality with this approach.

Reply:
We could use `getOrCreateWidget` at line 90.
But if we use `getOrCreateWidget`, the check whether a widget exists or not is executed twice.
I think the check result would be the same, but I didn't want to separate the check execution. (If there were "await" between the checks the result could change.)
I don't think I want to persist my solution,  so feel free to decline my opinion.